core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
> stay less contained, which is a strong disadvantage.
> I'm just mentioning this to explain my reasoning.

Thank you! 

I understand them and agree. My goal here is to make sure we do the same thing. For example, the current code doesn't check "http.enabled" and will enable port 9200 even if it's not needed (this specific example was very hard to catch because it wasn't even visible in the `NettyHttpServerTransport`). Same goes for `network.server` (which is probably bogus, but you get my point).

> Look at its static block, which then loads NettyUtils and then netty classes...

Argh. That's indeed very unfortunate.

I guess we should long term extract all of this to a NetworkSettings class of similar, but that is indeed a big refactoring and shouldn't be in the way of this.

I'm OK with proceeding with the current approach. Can I just ask that we make the settings keys static so we can use them to track all places where they are used? I'm afraid we'll make mistakes in future refactoring.

PS. I presume that just accessing a static const (like `NettyTransport.DEFAULT_PORT_RANGE` ) doesn't execute static coded?